src/main/java/service/impl/FruitServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package service.impl;
+
+import db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+import service.FruitService;
+import strategy.OperationHandler;
+
+public class FruitServiceImpl implements FruitService {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public FruitServiceImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void applyTransaction(FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {

```

Comment:
in cases like this, always invert if to throw exception first, and put rest of the logic after it - guard clause pattern. It will be easier to read since you don't need else after `throw`/`return`